### PR TITLE
Update maintainers list

### PR DESCRIPTION
**Pull request by peternied**
### Description
Update maintainers list

### Check List
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by davidlago — APPROVED** [review]

**Comment by peternied** [comment]
@opensearch-project/security could I get a second review?

**Review by cliu123 — APPROVED** [review]

**Pull request merged**
